rust/azure_iot_operations_connector/examples/base_connector_sample.rs

```diff
@@ -86,9 +86,95 @@ async fn run_program(
             }
         };
 
+        // Start handling the assets for this device endpoint
+        // if we didn't accept the inbound endpoint, then there's no reason to manage the assets
+        if endpoint_status.is_ok() {
+            tokio::task::spawn({
+                async move {
+                    run_assets(asset_creation_observation).await;
```

Comment:
Since run_assets already returns a `Future`, does this still need to be nested in an `async` block?

Reply:
good call, updated